src/System.Private.CoreLib/src/System/Threading/Monitor.cs

```diff
@@ -37,7 +37,7 @@ private static Lock GetLock(Object obj)
                 throw new ArgumentNullException("obj");
 
             if (Lock.IsLock(obj))
-                return (Lock)obj;
+                return RuntimeHelpers.UncheckedCast<Lock>(obj);
```

Comment:
Wouldn't it be better to have Lock.AsLock(obj) in the Lock class so that both the logic of IsLock and AsLock is optimized without moving that knowledge wherever this is needed?